week10/src/pages/Detail.jsx

```diff
@@ -0,0 +1,392 @@
+import { useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { GetPostDetailApi, EditPostApi, DeletePostApi } from "../api/post";
+import {
+    postHeart,
+    deleteHeart,
+    getHeartNum,
+    checkMyClick,
+} from "../api/heart";
+import { deleteComments, getComments, postComments } from "../api/comment";
+import { JWTtest } from "../api/user";
+import styled from "styled-components";
+
+import heart from "../img/heart.png";
+
+const Detail = () => {
+    const { postId } = useParams();
+    const [details, setDetails] = useState(null);
+    const [isEdit, setIsEdit] = useState(false);
+    const [post, setPost] = useState({ title: "", content: "" });
+    const [heartNum, setHeartNum] = useState(0);
+    const [myHeart, setMyheart] = useState(false);
+    const [myComments, setMyComments] = useState("");
+    const [commentsList, setCommentsList] = useState(null);
+    const [myAccount, setMyAccount] = useState("");
+    const nav = useNavigate();
+
+    useEffect(() => {
+        async function getDetails() {
+            const res = await GetPostDetailApi(postId);
+            setDetails(res);
+            const res2 = await JWTtest();
+
+            setMyAccount(res2.nickname);
+        }
+
+        async function getHeartNumInDetail() {
+            const num = await getHeartNum(postId);
+            setHeartNum(num);
+        }
+
+        async function getCommentsList() {
+            const res = await getComments(postId);
+            setCommentsList(res);
+        }
+
+        async function didIClickHeart() {
+            const HeartHistory = await checkMyClick(postId);
+            setMyheart(HeartHistory);
+        }
+
+        if (!localStorage.getItem("efubtoken")) {
+            alert("로그인 후 이용해주세요.");
+            nav("/register");
+        } else {
+            getDetails();
+            getHeartNumInDetail();
+            getCommentsList();
+            didIClickHeart();
+        }
+    }, [postId]);
+
+    useEffect(() => {
+        async function updateHeartNum() {
+            const num = await getHeartNum(postId);
+            setHeartNum(num);
+        }
+
+        updateHeartNum();
+    }, [myHeart, postId]);
+
+    if (!details) {
+        return <div>로딩 중...</div>;
+    }
+
+    const handleEditBtn = () => {
+        setIsEdit(true);
+    };
+
+    const onSubmitEditPost = async () => {
+        if (post.title.trim() === "") {
+            alert("제목을 입력해주세요.");
+        } else if (post.content.trim() == "") {
+            alert("내용을 입력해주세요.");
+        } else {
+            await EditPostApi(postId, post);
+            setIsEdit(false);
+            window.location.reload();
```

Comment:
```suggestion
            GetPostDetailApi(postId)
                .then(res => setDetails(res));
```
window reload 이벤트로 강제로 새로고침을 발생시키면 UX적 관점에서 부자연스러움이 있을 수 있습니다. 글 수정이 완료되고 나면 새로고침 대신 detail get API 함수를 다시 호출해주세요.
만약 위쪽 useEffect 문을 수정할 때 hook 내부 대신 외부에서 함수를 따로 정의했다면 해당 함수를 재사용 할 수 있겠죠?